pkg/kapp/clusterapply/add_or_update_change.go

```diff
@@ -292,6 +302,28 @@ func (c AddOrFallbackOnUpdateStrategy) Apply() error {
 	return c.aou.recordAppliedResource(createdRes)
 }
 
+type AddOrFallbackOnUpdateOrNoopStrategy struct {
+	newRes ctlres.Resource
+	aou    AddOrUpdateChange
+}
+
+func (c AddOrFallbackOnUpdateOrNoopStrategy) Op() ClusterChangeApplyStrategyOp {
+	return createStrategyFallbackOnUpdateOrNoopAnnValue
+}
+
+func (c AddOrFallbackOnUpdateOrNoopStrategy) Apply() error {
+	createdRes, err := c.aou.identifiedResources.Create(c.newRes)
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			c.aou.opts.allowNoopUpdates = true
```

Comment:
setting state on the object seems unnecessary -- we should just pass that into the function we are calling.